internals/secrethub/migrate.go

```diff
@@ -450,38 +434,79 @@ func (cmd *MigrateApplyCommand) Run() error {
 		}
 
 		for _, item := range vault.Items {
-			template := onepassword.NewItemTemplate()
-			for _, field := range item.Fields {
-				value, err := client.Secrets().ReadString(strings.TrimPrefix(field.Reference, secretReferencePrefix))
+			exists, err := onepassword.ExistsItemInVault(vault.Name, item.Name)
+			if err != nil {
+				return err
+			}
+			if !exists {
+				template := onepassword.NewItemTemplate()
+				for _, field := range item.Fields {
+					value, err := client.Secrets().ReadString(strings.TrimPrefix(field.Reference, secretReferencePrefix))
+					if err != nil {
+						return err
+					}
+					template.AddField(field.Name, value, field.Concealed)
+				}
+
+				err = onepassword.CreateItem(vault.Name, template, item.Name)
 				if err != nil {
 					return err
 				}
-				template.AddField(field.Name, value, field.Concealed)
-			}
-
-			err = onepassword.CreateItem(vault.Name, template, item.Name)
-			if err != nil {
-				return err
+			} else {
+				for _, field := range item.Fields {
+					hasField, err := onepassword.HasField(vault.Name, item.Name, field.Name)
+					if err != nil {
+						return err
+					}
+					if !hasField {
+						fmt.Fprintf(os.Stderr, "item %s.%s has missing field %s, please add this field manually to allow the migration tool to update it\n", vault.Name, item.Name, field.Name)
+						warningCount++
+						continue
+					}
+					value, err := client.Secrets().ReadString(strings.TrimPrefix(field.Reference, secretReferencePrefix))
+					if err != nil {
+						return err
+					}
+					opValue, err := onepassword.GetField(vault.Name, item.Name, field.Name)
+					if err != nil {
+						return err
+					}
+					if value != opValue {
+						if !cmd.update {
+							confirmed, err := ui.AskYesNo(cmd.io, fmt.Sprintf("The value of the field %s.%s from vault %s is different from its corresponding value in SecretHub. Do you want to set it to the value from SecretHub?", item.Name, field.Name, vault.Name), ui.DefaultYes)
+							if err != nil {
+								return fmt.Errorf("the value of the field %s.%s from vault %s is different from its corresponding value in SecretHub. Use --update to update all fields to their corresponding values from SecretHub", item.Name, field.Name, vault.Name)
+							}
+							if !confirmed {
+								fmt.Fprintln(cmd.io.Output(), "Skipped field.")
+								skippedCount++
+								continue
+							}
+						}
+						err = onepassword.SetField(vault.Name, item.Name, field.Name, value)
+						if err != nil {
+							return err
+						}
+						updatedCount++
+					} else {
+						alreadyUpToDateCount++
+					}
+				}
 			}
 		}
 		i++
 	}
-	fmt.Fprintln(cmd.io.Output(), "\nMigration completed successfully. You can now use these secrets from 1password.")
+
+	fmt.Fprintf(cmd.io.Output(), "\n%d fields were already up to date\n%d fields were updated\n%d fields were skipped\n", alreadyUpToDateCount, updatedCount, skippedCount)
```

Comment:
How does this message come across on the first run?

Reply:
Should we include newly created items?

Reply:
I believe we should use `are updated` or `updated` instead of `were updated`, since the updating still has effect, right?

```suggestion
	fmt.Fprintf(cmd.io.Output(), "%d items created\n", createdCount)
	fmt.Fprintf(cmd.io.Output(), "%d items updated\n", updatedCount)
	fmt.Fprintf(cmd.io.Output(), "%d items were already up-to-date\n", alreadyUpToDateCount)
	fmt.Fprintf(cmd.io.Output(), "%d items skipped\n", skippedCount)
```

Reply:
Yeah, that's right. Used the wrong verb tense here :sweat_smile: 

`updated` is indeed correct and more clear.

Reply:
And yes, we should include newly created items. But I believe we're counting fields not items, right?

Reply:
You're right, we should be counting fields :+1: